services/jenkins/jenkins-plugin-installs.service.js

```diff
@@ -0,0 +1,130 @@
+'use strict'
+
+const Joi = require('joi')
+
+const BaseJsonService = require('../base-json')
+const { NotFound, InvalidParameter } = require('../errors')
+const {
+  downloadCount: downloadCountColor,
+} = require('../../lib/color-formatters')
+const { metric } = require('../../lib/text-formatters')
+const { nonNegativeInteger } = require('../validators')
+
+const schemaInstallations = Joi.object()
+  .keys({
+    installations: Joi.object()
+      .required()
+      .pattern(nonNegativeInteger, nonNegativeInteger)
+      .min(1),
+  })
+  .required()
+
+const schemaInstallationsPerVersion = Joi.object()
+  .keys({
+    installationsPerVersion: Joi.object()
+      .required()
+      .pattern(Joi.string(), nonNegativeInteger)
+      .min(1),
+  })
+  .required()
+
+class JenkinsPluginInstalls extends BaseJsonService {
+  async fetch({ plugin, version }) {
+    const url = `https://stats.jenkins.io/plugin-installation-trend/${plugin}.stats.json`
+    const schema = this.constructor._getSchema(version)
+    return this._requestJson({ url, schema })
+  }
+
+  static render({ label, installs }) {
+    return {
+      label: label,
+      message: metric(installs),
+      color: downloadCountColor(installs),
+    }
+  }
+
+  static _getSchema(version) {
+    if (version) {
+      return schemaInstallationsPerVersion
+    } else {
+      return schemaInstallations
+    }
+  }
+
+  static _getLabel(version) {
+    if (version) {
+      return 'installs@' + version
+    } else {
+      return 'installs'
+    }
+  }
+
+  async handle({ info, plugin, version }) {
+    const label = this.constructor._getLabel(version)
+    const json = await this.fetch({ plugin, version })
+
+    let installs
+    if (info === 'iv') {
+      installs = json.installationsPerVersion[version]
+      if (!installs) {
+        throw new NotFound({
+          underlyingError: new Error('non-existent version'),
+        })
+      }
+    } else if (info === 'i') {
+      const latestDate = Object.keys(json.installations)
+        .sort()
+        .slice(-1)[0]
+      installs = json.installations[latestDate]
+    } else {
+      throw new InvalidParameter({
```

Comment:
Can this case actually ever be reached? Shouldn't the regex guarantee that we can only have either `iv` or `i`?